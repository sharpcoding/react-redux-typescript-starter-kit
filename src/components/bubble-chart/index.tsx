import * as d3 from 'd3';
import * as _ from 'lodash';
import * as React from 'react';
import { DotPlotter } from './dot-plotter';
import { IBubblePoint } from './model/bubble-point';

export { IBubblePoint };

export interface IBubbleChartProps {
  width: number;
  height: number;
  backgroundColor: string;
  dots: IBubblePoint[];
}

export const BubbleChart = (props: IBubbleChartProps) => {
  const getCssStyle = () => {
    return { backgroundColor: props.backgroundColor  };
  };

  const xScale = d3.scaleLinear()
    .domain([_.isEmpty(props.dots) ? 0 : _.minBy(props.dots, (d: IBubblePoint) => d.x).x,
             _.isEmpty(props.dots) ? 0 : _.maxBy(props.dots, (d: IBubblePoint) => d.x).x])
    .range([0, props.width]);

  const yScale = d3.scaleLinear()
    .domain([_.isEmpty(props.dots) ? 0 : _.minBy(props.dots, (d: IBubblePoint) => d.y).y,
             _.isEmpty(props.dots) ? 0 : _.maxBy(props.dots, (d: IBubblePoint) => d.y).y])
    .range([0, props.width]);

  return (
    <svg
      width={props.width}
      height={props.height}
      style={getCssStyle()}>
      <DotPlotter xScale={xScale} yScale={yScale} dots={props.dots} />
    </svg>
  );
};