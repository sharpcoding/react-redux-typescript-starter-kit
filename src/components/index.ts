export { BubbleChart } from './bubble-chart';
export { IBubblePoint } from './bubble-chart/model/bubble-point';