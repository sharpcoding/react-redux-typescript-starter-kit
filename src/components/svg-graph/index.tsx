import * as React from "react";

export interface ISvgGraphProps {
  width: number; 
  height: number;
  backgroundColor: string;
}

export const SvgGraph = (props: ISvgGraphProps) => {
  let getCssStyle = () => {
    return { backgroundColor: props.backgroundColor  };
  }
  
  return (
    <svg 
      width={props.width} 
      height={props.height}
      style={getCssStyle()} >
    </svg>
  );
}