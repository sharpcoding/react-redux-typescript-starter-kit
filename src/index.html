<!DOCTYPE html>
<html>
<head>
    <title>React/Redux/TypeScript/D3/Bootstrap Starter Kit</title>
    <meta http-equiv="content-type" content="text/html; charset=UTF8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />
</head>
<body>
    <div id="app"></div>
    <script src="../dist/compiled.bundle.js"></script>
</body>
</html>