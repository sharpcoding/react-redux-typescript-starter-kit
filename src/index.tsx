import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import '../assets/scss/styles.scss';
import { BubbleChartScreen } from '@screens/bubble-chart-screen';
import { store } from '@store/store-creator';

ReactDOM.render(
  <Provider store={store}>
    <span>
      <h3>Hello from React, Redux, Typescript and d3</h3>
      <BubbleChartScreen />
    </span>
  </Provider>,
  document.body.appendChild(document.createElement('div')),
);