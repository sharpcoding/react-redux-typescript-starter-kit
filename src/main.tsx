import './css/app.css';
import 'bootstrap/dist/css/bootstrap.css';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Store, createStore, compose, applyMiddleware, combineReducers } from 'redux';
import { Provider } from 'react-redux';

// import './styles.scss';
// import * as React from 'react';
// import * as ReactDOM from 'react-dom';
// import thunk from 'redux-thunk';
// import { hpTimeSeriesChartCsvLoadingActionCreators } from '../hp-time-series-chart/csv-loading/action-creators';
// import { HashRouter, Route } from 'react-router-dom';
// import { Store, createStore, compose, applyMiddleware, combineReducers } from 'redux';
// import { Provider } from 'react-redux';
// import { HpTimeSeriesChart } from '../hp-time-series-chart';
// import { GraphScreen } from './demo-container';
// import { storeCreator } from './demo-container/store-creator';

// const combinedReducers = combineReducers({
//   chartState: storeCreator
// });

// //this is the callback function required in order to have this Chrome extension https://github.com/zalmoxisus/redux-devtools-extension working
// const composeEnhancers = window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] || compose;
// const store: Store<any> = createStore(combinedReducers, composeEnhancers(), applyMiddleware(thunk));

// ReactDOM.render(
//   <Provider store={store}>
//     <div>
//       <GraphScreen />
//     </div>
//   </Provider>,
//   document.body.appendChild(document.createElement('div'))
// );

const combinedReducers = combineReducers({
  // graphScreenState
});

//this is the callback function required in order to have this Chrome extension https://github.com/zalmoxisus/redux-devtools-extension working
const composeEnhancers = window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] || compose;
const store: Store<any> = createStore(combinedReducers, composeEnhancers(applyMiddleware()));

ReactDOM.render(
  <Provider store={store}>
    <span>
      <h3>Hello from React, Redux, D3 and Bootstrap</h3>
      {/* <GraphMainScreen /> */}
    </span>
  </Provider>,
  document.getElementById('app')
);