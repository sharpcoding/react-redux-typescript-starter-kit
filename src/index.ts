import Bootstrap from 'bootstrap/dist/css/bootstrap.css';
import './main';