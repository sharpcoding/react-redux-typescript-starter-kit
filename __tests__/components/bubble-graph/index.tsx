import * as dateFns from 'date-fns';
import * as _ from 'lodash';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BubbleChart } from '../../../src/components';

describe('Bubble Chart Component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BubbleChart backgroundColor='red' dots={[]} width={1024} height={768} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});