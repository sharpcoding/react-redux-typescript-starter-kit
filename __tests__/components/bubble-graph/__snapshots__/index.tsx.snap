// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bubble Chart Component renders correctly 1`] = `
<svg
  height={768}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
  width={1024}
>
  <g />
</svg>
`;
